pallets/order-book/src/lib.rs

```diff
@@ -582,6 +592,98 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Fill an existing order, based on the provided ratio.
+		#[pallet::call_index(7)]
+		#[pallet::weight(T::Weights::fill_order_partial())]
+		pub fn fill_order_partial(
+			origin: OriginFor<T>,
+			order_id: T::OrderIdNonce,
+			fulfillment_ratio: Perquintill,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+			let order = <Orders<T>>::get(order_id)?;
+
+			ensure!(
+				!fulfillment_ratio.is_zero(),
+				Error::<T>::InvalidPartialOrderRatio
+			);
+
+			let partial_buy_amount = fulfillment_ratio.mul_floor(order.buy_amount);
+			let partial_sell_amount = fulfillment_ratio.mul_floor(order.max_sell_amount);
+			let remaining_buy_amount = order
+				.buy_amount
+				.checked_sub(&partial_buy_amount)
+				.ok_or(Error::<T>::RemainingBuyAmountError)?;
+			let partial_fulfillment = !remaining_buy_amount.is_zero();
+
+			ensure!(
+				partial_buy_amount >= order.min_fulfillment_amount,
```

Comment:
This would mean the `min_fullfillment_amount` is not a global min but rather a `min_per_buy`. I am okay with that, just wanted to channel this by you @hieronx .

Reply:
Please let me know if we are OK with this, otherwise we can add a check for the remaining buy amount to ensure that it doesn't go too low, even though I'm not sure if that makes sense ^^

Reply:
No, I was rather thinking that it is a `one-time` amount. Like, I wanna get at least `x`. But it is fine to go with the given solution.